robot/basestation/static/js/science.js

```diff
@@ -490,78 +537,215 @@ $(document).ready(function () {
+
   $('#drill-stop-btn').click(function (msgs) {
     if (!isScienceActivated()) {
       return
     }
 
     // drill stop
-    sendRequest("Science", 'ds', function (msgs) {
+    sendRequest('Science', 'ds', function (msgs) {
       console.log('msgs', msgs)
 
       if (msgs[1].includes('ds done')) {
         appendToConsole('Success')
         lightUp('#drill-stop-btn')
         greyOut('#drill-max-speed-go-btn')
+        greyOut('#set-speed-go-btn')
+        greyOut('#set-time-go-btn')
       } else {
         appendToConsole('Something went wrong')
       }
     })
   })
 
-  $('#set-speed-go-btn').click(function (msgs) {
+  $('#elevator-max-speed-go-btn').click(function (msgs) {
     if (!isScienceActivated()) {
       return
     }
 
-    console.log('msgs', msgs)
+    sendRequest('Science', 'ego', function (msgs) {
+      console.log('msgs', msgs)
 
-    if (msgs[1].includes('drillspeed done')) {
-      appendToConsole('Success')
-      lightUp('#set-speed-go-btn')
-      greyOut('#drill-max-speed-go-btn')
-    } else {
-      appendToConsole('Something went wrong')
+      if (msgs[1].includes('ego done')) {
+        appendToConsole('Success')
+        lightUp('#elevator-max-speed-go-btn')
+        greyOut('#elevator-stop-btn')
+      } else {
+        appendToConsole('Something went wrong')
+      }
+    })
+  })
+
+  $('#set-feed-go-btn').click(function (msgs) {
+    if (!isScienceActivated()) {
+      return
+    }
+
+    let elevatorFeed = $('#elevator-feed').val()
+    let cmd = 'ef ' + elevatorFeed
+    let requestedFeed = Number(elevatorFeed)
+
+    // invalid range check
+    // values under 10% don't actually move the elevator
+    if (requestedFeed < 10 || requestedFeed > 100) {
+      color('#elevator-feed', 'orange')
+      appendToConsole('Valid ranges for elevator feed: [10, 100]')
+      return
     }
+
+    color('#elevator-feed', 'white')
+
+    sendRequest('Science', cmd, function (msgs) {
+      console.log('msgs', msgs)
+
+      if (msgs[1].includes('elevatorfeed done')) {
+        appendToConsole('Success')
+        lightUp('#set-feed-go-btn')
+        greyOut('#elevator-stop-btn')
+      } else {
+        appendToConsole('Something went wrong')
+      }
+    })
   })
 
-  $('#set-time-go-btn').click(function (msgs) {
+  // elevator - set distance command
+  $('#set-distance-go-btn').click(function (msgs) {
     if (!isScienceActivated()) {
       return
     }
 
-    console.log('msgs', msgs)
+    let elevatorDistance = $('#elevator-distance').val()
+    let cmd = 'elevatordistance ' + elevatorDistance
+    let requestedDistance = Number(elevatorDistance)
+
+    // invalid range check
+    // if the system on the MCU were closed loop we would do it based off current position
+    // otherwise we assume there will be limit switches for protection
+    if (requestedDistance <= 0 || requestedDistance > 15) {
+      color('#elevator-distance', 'orange')
+      appendToConsole('Valid ranges for elevator feed: (0, 15]')
```

Comment:
I would be explicit rather than using this cause I forget which of ( and ] means inclusive and exclusive

Reply:
Quick mafs. ( is exclusive, ] is inclusive. The user should safely be able to figure it out via trial and error if they forget.